.envrc

```diff
@@ -104,31 +104,36 @@ done
 
 ### Python ###
 
-info "Checking virtualenv..."
-
-# direnv set -u's, so we need to do this.
-VIRTUAL_ENV="${VIRTUAL_ENV:-}"
+info "Activating virtualenv..."
 
-if [ -n "$VIRTUAL_ENV" ]; then
-    # we're enforcing that virtualenv be in .venv, since future tooling e.g. venv-update will rely on this.
-    if [ "$VIRTUAL_ENV" != "${PWD}/.venv" ]; then
-        info "You're in a virtualenv, but it's not in the expected location (${PWD}/.venv)"
-        advice_init_venv
-    fi
-else
-    if [ ! -f ".venv/bin/activate" ]; then
-        info "You don't seem to have a virtualenv."
-        advice_init_venv
-    fi
+# we're enforcing that virtualenv be in .venv, since future tooling e.g. venv-update will rely on this.
+if [ ! -f ".venv/bin/activate" ]; then
+    info "You don't seem to have a virtualenv."
+    advice_init_venv
 fi
 
-info "Activating virtualenv..."
+# The user might be cd'ing into sentry with another non-direnv managed
+# (in that it would be automatically deactivated) virtualenv active.
+deactivate 2>/dev/null || true
+
 source .venv/bin/activate
 # XXX: ideally, direnv is able to export PS1 as modified by sourcing venvs
 #      but we'd have to patch direnv, and ".venv" isn't descriptive anyways
 unset PS1
 [ "$(command -v python)" != "${PWD}/.venv/bin/python" ] && die "Failed to activate virtualenv."
 
+# direnv set -u's, so we need to do this.
+VIRTUAL_ENV="${VIRTUAL_ENV:-}"
```

Comment:
would VIRTUAL_ENV ever be set if we did deactivate?

Reply:
Guaranteed to be set by the `source .venv/bin/activate` above.

Reply:
Actually, this is way better: https://github.com/getsentry/sentry/pull/17556/commits/203f108cea2bd2f0a4f6b9132ffbc8e85d270af3